sae_lens/training/activations_store.py

```diff
@@ -234,7 +234,7 @@ def __init__(
 
         self.n_dataset_processed = 0
 
-        self.estimated_norm_scaling_factor = 1.0
+        self.estimated_norm_scaling_factor = None
```

Comment:
Would love thoughts on this, I've changed this to initialise as `None` so that attempting to use it before explicitly setting it raises an error, which (as far as I can tell) would seem good as that would be an error case.

Unfortunately, `ActivationsStore.get_buffer` relies on `self.estimated_norm_scaling_factor` by way of `self.apply_norm_scaling_factor`. So we need it to have a value when loading the batches used to estimate the scaling factor (see comments below). My solution of temporarily setting and unsetting it is pretty ugly, so it might be best to simply keep this initialising as `1.0` and accept the aforementioned risk.

(I did some wider refactoring which got around this, but I decided not to submit them as they made this PR bigger and wider-reaching.)